flax/errors.py

```diff
@@ -423,6 +423,26 @@ def __call__(self, x):
 
     variables = SomeModule(features=3).init(random.PRNGKey(0), jnp.ones((1, )))
 
+  Alternatively, if you need to introduce canonicalization or automatically derive
+  other attributes, you can add logic to `__post_init__`:
+
+    class Foo(nn.Module):
```

Comment:
This only works if `canonicalize(canonicalize(orig_attributes)) == canonicalize(orig_attributes)` otherwise cloning can cause side-effects. @avital has a PR that stores the initial attr values to solve this case (and enables atttribute updates in setup)

Reply:
ah right of course,  I'll axe the text in the docs for this PR then - I still think the logic of this PR is an improvement on the setattr jumble that we had before and should be compatible with any such further fixes.

Reply:
note that the motivating case was one of deriving other attributes from dataclass attribs rather than normalizing dataclass attribs directly.

Reply:
@avital - you have a link to that PR?